### PR TITLE
GSoC 2022: Nitish Chauhan Week 2

### PR DESCRIPTION
**Pull request by nitishchauhan0022**
Changes proposed in this pull request:
- [pgr_hawickCircuits] [doc] adding CMakeLists.txt
- [pgr_hawickCircuits] [include] adding c_type circuits_rt.h
- [pgr_hawickCircuits] [src] modifying hawickCircuits.c for arrays
- [pgr_hawickCircuits] Doing some requested changes

### CODE REVIEW

File: include/drivers/circuits/hawickcircuits_driver.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#endif
+
+typedef struct Edge_t Edge_t;
+typedef struct circuits_rt circuits_rt;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************************************************
```
Comment (cvvergara) [inline include/drivers/circuits/hawickcircuits_driver.h:48]: This comment is also an inheritance of 2.0.
Please remove.
It is annoying that if you change the sql file, then you have to look for all the comments to fix this.
It is better not to have it.
In any case if you modify the SQL, the compiler will let you know what needs to be changed.

If you really need that reminder of how the SQL looks like, then while developing, you could have a link to:
https://github.com/pgRouting/GSoC-pgRouting/blob/nitish-2022/sql/circuits/_hawickCircuits.sql
but if you do, open an issue in your fork to remind you that this link has to be deleted before merging to pgRouting.

File: include/drivers/circuits/hawickcircuits_driver.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*PGR-GNU*****************************************************************
+File: circuits_driver.h
+
+Generated with Template by:
+Copyright (c) 2022 pgRouting developers
+Mail: project@pgrouting.org
+
+Function's developer:
+Copyright (c) 2022 Nitish Chauhan
+Mail: nitishchauhan0022@gmail.com
```
Comment (cvvergara) [inline include/drivers/circuits/hawickcircuits_driver.h:10]: Just for security reasons
nitishchauhan0022@gmail.com -> nitishchauhan0022 at gmail.com

**Review by cvvergara — CHANGES_REQUESTED** [review]

File: include/drivers/circuits/hawickcircuits_driver.h (hunk 2 above)
Comment (nitishchauhan0022) [reply include/drivers/circuits/hawickcircuits_driver.h:10]: ok, I will make all the changes.

**Comment by nitishchauhan0022** [comment]
Ok, I will make all the changes.

**Pull request merged**